### PR TITLE
Add auto_device_count and device name support

### PR DESCRIPTION
**Pull request by jerome-habana**
Signed-off-by: Jerome <janand@habana.ai>

## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/accelerators/hpu.py [hunk 1]
```diff
@@ -52,13 +55,27 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
 
     @staticmethod
     def auto_device_count() -> int:
-        """Get the devices when set to auto."""
-        # TODO(@kaushikb11): Update this when api is exposed by the Habana team
-        return 8
+        """Returns the number of HPU devices when the devices is set to auto."""
+        try:
+            return torch_hpu.device_count()
```
Comment (kaushikb11) [inline src/pytorch_lightning/accelerators/hpu.py:60]: What versions have this?

File: src/pytorch_lightning/accelerators/hpu.py [hunk 2]
```diff
@@ -52,13 +55,27 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
...
+        except AttributeError:
+            return 0
```
Comment (kaushikb11) [inline src/pytorch_lightning/accelerators/hpu.py:62]: If not all versions have this and Habana is available, it should be 8, right?

File: src/pytorch_lightning/accelerators/hpu.py (hunk 1 above)
Comment (jerome-habana) [reply src/pytorch_lightning/accelerators/hpu.py:60]: With synapse release 1.5, we have this interface enabled

File: src/pytorch_lightning/accelerators/hpu.py (hunk 2 above)
Comment (jerome-habana) [reply src/pytorch_lightning/accelerators/hpu.py:62]: 1 node contains 8 cards

File: src/pytorch_lightning/accelerators/hpu.py (hunk 1 above)
Comment (kaushikb11) [reply src/pytorch_lightning/accelerators/hpu.py:60]: This above logic would be better:

```python
if _HPU_VERSION_X:
	return torch_hpu.device_count()
return 8
```

File: src/pytorch_lightning/CHANGELOG.md [hunk 3]
```diff
@@ -122,6 +122,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - The `WandbLogger` will now use the run name in the logs folder if it is provided, and otherwise the project name  ([#12604](https://github.com/PyTorchLightning/pytorch-lightning/pull/12604))
 
 
+- Updated hpus's auto_device_count, device available & name usage based on the updated torch habana package ([#13423](https://github.com/PyTorchLightning/pytorch-lightning/pull/13423))
```
Comment (kaushikb11) [inline src/pytorch_lightning/CHANGELOG.md:125]:
```suggestion
- Updated Habana Accelerator's `auto_device_count`, `is_available` & `get_device_name` methods based on the latest torch habana package ([#13423](https://github.com/PyTorchLightning/pytorch-lightning/pull/13423))
```

**Review by kaushikb11 — APPROVED** [review]

File: src/pytorch_lightning/accelerators/hpu.py [hunk 4]
```diff
@@ -52,13 +55,28 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
...
+        except AttributeError:
+            rank_zero_debug("HPU auto_device_count failed, returning default count of 8.")
+            return 8
 
     @staticmethod
     def is_available() -> bool:
-        return _HPU_AVAILABLE
+        """Returns a bool indicating if HPU is currently available."""
+        try:
+            return torch_hpu.is_available()
```
Comment (carmocca) [inline src/pytorch_lightning/accelerators/hpu.py:69]: What's the difference between `torch_hpu.is_available` and `_HPU_AVAILABLE = habana_frameworks.torch.utils.library_loader.is_habana_avaialble()`?

Should this still use `_HPU_AVAILABLE` and instead update it's definition in `imports.py`?

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/accelerators/hpu.py (hunk 4 above)
Comment (jerome-habana) [reply src/pytorch_lightning/accelerators/hpu.py:69]: _HPU_AVAILABLE checks for the availability of the packages and drivers ensuring device can be uses and is_habana_available() is post creation of a synapse session to make queries to the library like device_count, name etc

File: src/pytorch_lightning/accelerators/hpu.py (hunk 1 above)
Comment (jerome-habana) [reply src/pytorch_lightning/accelerators/hpu.py:60]: I see your point but this is just for sanity

File: tests/tests_pytorch/accelerators/test_hpu.py [hunk 5]
```diff
@@ -40,6 +40,11 @@ def test_availability():
     assert HPUAccelerator.is_available()
 
 
+@RunIf(hpu=True)
+def test_device_name():
+    assert HPUAccelerator.get_device_name() in ["GAUDI"]
```
[the remark this answers is not in the document]
Comment (jerome-habana) [reply tests/tests_pytorch/accelerators/test_hpu.py]: This was kept to support the next generations
